src/pandapipes/component_models/heat_consumer_component.py

```diff
@@ -201,9 +202,10 @@ def adaption_after_derivatives_thermal(cls, net, branch_pit, node_pit, idx_looku
         hc_pit = branch_pit[f:t, :]
         consumer_array = get_component_array(net, cls.table_name(), mode='heat_transfer')
 
-        # Any MODE where TRETURN is given
-        mask = consumer_array[:, cls.MODE] == cls.QE_TR
+        mask= consumer_array[:, cls.MODE] == cls.QE_TR
         if np.any(mask):
+            mask_ign = hc_pit[:, QEXT] == 0
+            mask = mask & ~mask_ign
```

Comment:
In this case, the outlet temperature would equal to the inlet temperature, right?

Reply:
Yes!